pages/api/utils/policy-assignment-utils.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2021 The Rode Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getPolicyByPolicyId } from "./policy-utils";
+
+const getPolicyIdAndVersion = (policyVersionId) => {
+  const [policyId, policyVersion] = policyVersionId.split(".");
+
+  return {
+    policyId,
+    policyVersion,
+  };
+};
+
+export const mapToClientModelWithPolicyDetails = async (assignment) => {
+  const { policyId, policyVersion } = getPolicyIdAndVersion(
+    assignment.policyVersionId
+  );
+
+  const { data, error } = await getPolicyByPolicyId(policyId);
+
+  if (error) {
+    throw Error(
+      `Error while fetching policy ${policyId} for policy group assignments.`
+    );
+  }
+
+  return {
+    ...assignment,
+    policyId,
+    policyVersion,
```

Comment:
If this is fetching the policy, couldn't get the version from the response? I see that it's [not mapped](https://github.com/rode/rode-ui/blob/79ddca75fea4ac2de1cf209bb1da84b89f08f49d/pages/api/utils/policy-utils.js#L29-L36) at the moment, but I was hoping to avoid parsing the policy version id in the UI

Reply:
I think this parsing is still going to be required here because the assignment doesn't have the policyID attached, and calling to get the policy might return a different version than the one currently assigned to the policy group.

Reply:
`GetPolicy` can also take the policy version id, and return the policy and that particular version instead of whatever the current version is.

Reply:
Ooooo okay awesome, then yeah I will update this stuff 👍